cli/options.go

```diff
@@ -37,53 +37,84 @@ type ProjectOptions struct {
 	Name        string
 	WorkingDir  string
 	ConfigPaths []string
-	Environment []string
+	Environment map[string]string
```

Comment:
Haven't looked closely at how it's used, so this is just a comment (in case it's important); note that on the containers itself, environment variables are set as a `[]string`. IIRC, one of the reasons was that order may be important (and Go maps randomise order when iterating over them).

For example; this is valid, and later values for `FOO` can override earlier values;

```
docker run -e FOO=bar -e FOO=baz -e BLA=BLA
```

If there's an intermediate map, we may loose the original.

Depending on "when" env-vars are expanded, and which values should be taken into account, `BLA=$FOO` may have to be substituted, and same for `-e BLA` (which takes the value of `$BLA` from the current environment (if present). Without looking deeply into the code, this may mean that we may have to distinguish `nil` (`-e FOO`) from `empty` values (`-e FOO=` / `-e FOO=""`).

Reply:
The code compose-go was inherited from also eventually uses a `map[string]string` after passing CLI options as a `[]string`: https://github.com/docker/cli/blob/master/cli/compose/types/types.go#L63

Reply:
Thx! So I guess if there _is_ a problem in that approach, it's not something _new_ 👍